prov/efa/src/rxr/rxr_read.c

```diff
@@ -155,7 +155,14 @@ struct rxr_read_entry *rxr_read_alloc_entry(struct rxr_ep *ep, int entry_type, v
 		}
 	} else {
 		assert(lower_ep_type == SHM_EP);
-		memset(read_entry->mr, 0, read_entry->iov_count * sizeof(struct fid_mr *));
+		/*
+		 * If CMA is NOT available, FI_MR_VIRT_ADDR will be cleared in shm provider,
+		 * use 0-based offset for shm's RMA operation.
+		 */
+		if (!rxr_env.shm_cma_cap) {
+			for (i = 0; i < read_entry->rma_iov_count; i++)
+				read_entry->rma_iov[i].addr = 0;
+		}
```

Comment:
please move the change of rma_iov to the next part, and the line of ```memset``` should not be removed.

Reply:
@wzamazon and I think it's better to let shm expose FI_MR_VIR_ADDR bit regardless of CMA capability of system, so we don't need to deal with 0-based offset.